### PR TITLE
XSendFileTemporary

### PR DESCRIPTION
**Pull request by jrhee**
Heres what I added:

On the first commit:
  If the xsendfile filter does not find the header 'X-SendFile', it looks for 'X-SendFile-Temporary'.  If 'X-Send-File-Temporary' is found, the apr_file_open call includes APR_DELONCLOSE.

On the second commit:
  I realized that developers could potentially be stupid and delete an important file on accident.  Thus, I added a new configuration command:  'XSendFileTemporaryPath /path/to/temp/dir'.  Files can be sent from both XSendFilePath and XSendFileTemporaryPath, but only deleted from XSendFileTemporaryPath.  Any calls to X-SendFile-Temporary on a path that is not a child of XSendFileTemporaryPath will fail.  This way, users can designate a temporary-file-directory' without risking accidental file deletion from a non-temporary-directory.  

Look it over and let me know what you think.  The whole purpose of me writing this is so that you can release it, support it, and other people will be able to use it.  So...let me know if you find anything that would prevent you from releasing this, and hopefully we can resolve any issues.

### CODE REVIEW

**Comment by nmaier** [comment]
Thanks for the pull request so far.
Would be great if you could address my comments.

Another nit not mentioned yet: Please use two spaces instead of tabs for indentation.

**Comment by jrhee** [comment]
I made all the changes you requested.  'AllowFileDelete' is now an option on XSendFilePath.  paths are now stored in the structure xsendfile_path_t which contains the char\* path and an is_temp flag, simplifying the management of permissions.

File: mod_xsendfile.c [hunk 1]
```diff
@@ -225,10 +235,9 @@ static apr_status_t ap_xsendfile_get_filepath(request_rec *r, xsendfile_conf_t *
   ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, r->server, "xsendfile: path is %s", root);
 #endif
 
-  /* merge the array */
-  if (root) {
-    patharr = apr_array_make(r->pool, conf->paths->nelts + 1, sizeof(char*));
-    *(const char**)(apr_array_push(patharr)) = root;
+  if (!is_temp && root) {
```
Comment (jrhee) [inline mod_xsendfile.c:238]: I'm not really sure about the use case for 'root', so I made the assumption that we only ever allow deletion from paths specified in the config.

**Comment by nmaier** [comment]
Ok, I rebased that into a single commit, fixed some build issues (MSVC) and style stuff.
Also added you to contributors: 82f0ed2de41b7c38d7cf1724b23540e344dd2243 (let me know if you want your email address included)
And added some docs: f6b853ce0e555b61f83f928d9f927349346018b4

**Pull request closed without merging**